### PR TITLE
fix(list): #MCDT-584 fix not display seance with only screen

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
When a session has only one image in description it does not appear in list.
Basically we filter all the html tags to retrieve only the text of the description. Then we check the length of the text. The img tag being filter the length of the text was empty and was not displayed in the list.
By modifying the filter regex to select the img tag, the length is no longer empty and the session is displayed.

## Checklist tests
Create a session with only one image, check if it is displayed in the list.

## Issue ticket number and link
https://entsupport.gdapublic.fr/browse/MCDT-584

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [x] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/directives/async-autocomplete.ts [hunk 1]
```diff
@@ -63,7 +64,7 @@ export const asyncAutocomplete = ng.directive('asyncAutocomplete', ['$timeout',
             $scope.match = newVal;
             cancelAnimationFrame(token);
             setLoadingStatus(false);
-            $scope.$apply();
+            DateUtils.safeApply($scope).catch(e => console.error(e));
```
Comment (BastienBenard) [inline src/main/resources/public/ts/directives/async-autocomplete.ts:67]: Utilisation d'un safeApply pour éviter des erreur de digest

File: src/main/resources/public/ts/directives/async-autocomplete.ts [hunk 2]
```diff
@@ -74,7 +75,7 @@ export const asyncAutocomplete = ng.directive('asyncAutocomplete', ['$timeout',
         const closeDropDown = function () {
             setLoadingStatus(false);
             $scope.match = undefined;
-            $scope.$apply();
+            DateUtils.safeApply($scope).catch(e => console.error(e));
```
Comment (BastienBenard) [inline src/main/resources/public/ts/directives/async-autocomplete.ts:78]: Utilisation d'un safeApply pour éviter des erreur de digest

File: src/main/resources/public/ts/model/__mocks__/entcore.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import {Moment} from "moment";
```
Comment (BastienBenard) [inline src/main/resources/public/ts/model/__mocks__/entcore.ts:1]: fichier utiliser pour les tests

File: src/main/resources/public/ts/directives/async-autocomplete.ts (hunk 2 above)
Comment (JinDouang) [reply src/main/resources/public/ts/directives/async-autocomplete.ts:78]: Est-ce que tu peux créer une nouvelle fonction safeApply que tu exportes partout en tant que fonction au lieu qu'il passe par une DateUtil ?

File: src/main/resources/public/ts/directives/async-autocomplete.ts (hunk 1 above)
Comment (JinDouang) [reply src/main/resources/public/ts/directives/async-autocomplete.ts:67]: Pas besoin du catch
Quoi qu'il arrive t'auras une erreur si c est le cas

File: src/main/resources/public/ts/directives/async-autocomplete.ts [hunk 4]
```diff
@@ -50,7 +51,7 @@ export const asyncAutocomplete = ng.directive('asyncAutocomplete', ['$timeout',
 
         const setLoadingStatus = (status: boolean = true) => {
             $scope.loading = status;
-            $scope.$apply();
+            DateUtils.safeApply($scope).catch(e => console.error(e));
```
[the remark this answers is not in the document]
Comment (JinDouang) [reply src/main/resources/public/ts/directives/async-autocomplete.ts:54]: Pas besoin du catch
Quoi qu'il arrive t'auras une erreur si c est le cas

**Review by JinDouang — COMMENTED** [review]
Possibilité de recréer fonction safeApply en exportant la fonction seulement ?

File: src/main/resources/public/ts/utils/dateUtils.ts [hunk 5]
```diff
@@ -81,21 +81,9 @@ export class DateUtils {
         return xhtml;
     }
 
-    static safeApply(that) {
```
Comment (BastienBenard) [inline src/main/resources/public/ts/utils/dateUtils.ts:84]: Déplacé dans un function exporter

File: src/main/resources/public/ts/directives/async-autocomplete.ts [hunk 6]
```diff
@@ -1,4 +1,6 @@
 import {$, idiom as lang, ng} from 'entcore';
+import {DateUtils} from "../utils/dateUtils";
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/async-autocomplete.ts:2]: tjr util import ?

**Pull request merged**